torch/distributed/tensor/parallel/api.py

```diff
@@ -255,16 +266,23 @@ def _parallelize_linear(
         A :class:`nn.Module` object parallelized.
     """
 
-    if not isinstance(module, nn.Linear):
+    if not isinstance(module, (nn.Linear, nn.Embedding)):
         raise RuntimeError(
-            f"Expect a torch.nn.Linear module but received {type(module)}!"
+            f"Expect a torch.nn.Linear or a torch.nn.Embedding module but received {type(module)}!"
         )
 
     if not isinstance(parallel_style, ParallelStyle):
         raise RuntimeError(
             "Expect a ParallelStyle object but received" f" {type(parallel_style)}!"
         )
 
+    if isinstance(module, nn.Embedding) and not isinstance(
```

Comment:
that's why I feel that we should make the `_partition_fn` (private first before exposing to user) to each parallel style, so that each style can just check what they supports separately and error out if they are not those nn.Modules..

But we can do this later

Reply:
Right, we can discuss about this design later. If we can get the discussion converged, we can start implementing it. To be that might be a big change, and I want to collect more users feedback for that.

Reply:
what I mean is that there's no user feedback needed first, we are not opening this to users, that defnitely need to have some discussion on design. But first I think we can make it private and it will make our TP code much more clean than the current state.

Reply:
What I was trying to say is that even adding `_partition_fn ` might need more discussions. Or you feel we should not add logics here in TP API?

Reply:
I don't quite understand why this need discussions as it's not user facing? I think we should not continue adding the logic in TP API here, the current way of handling module partitioning becomes pretty messy as the module logic continue be added to the API itself, if more modules need to be partitioned in the future, the API will become cumbersome. It is already today, why Embedding is linear like? they are totally different kind of modules

Reply:
Well I guess our previous conclusion is:
1. For now we mostly focus on transformer so that we down-scope this API a little bit? And we are ok that we add embedding logic in the TP API as well?
2. If we see new use cases down the road, we can see how we can extend TP API to accommodate to new use cases. 

Or we now want to consider more generic cases? Adding `_partition_fn` to me is indeed user-facing because we are changing behaviors and is there a use case which the current TP API does not cover?

Regarding the code cumbersomeness, yes, I am thinking of refactoring the TP API a little bit. Also, linear and embedding are two different modules, but the way they are sharded (colwise) is the same, that's why I want some code to be reused here.